src/views/modals/TaskModal.svelte

```diff
@@ -0,0 +1,65 @@
+<Modal size="fs" bind:open on:close={closeModal}>
+	<h3 slot="header">{@html dataSource?.name}</h3>
+	{#if periodicTable}
+		<div class="content periodic-table">
+			<Autocomplete {predefined} bind:selected placeholder="Type or select 3 elements" />
+			<PeriodicTable bind:selected bind:clear />
+			<div class="buttons">
+				<Button size="md" loading={refinementStarted} on:click={startRefinement}
+					>Start refinement</Button
+				>
+			</div>
+		</div>
+	{:else}
+		<div class="content">
+			<DataView dataSourceId={dataSource?.id} />
+		</div>
+	{/if}
+	<p slot="footer">
+		<Button size="md" on:click={showPeriodicTable}>Show Periodic Table</Button>
+		<Button size="md" on:click={closeModal}>Close</Button>
+	</p>
+</Modal>
+
+<script lang="ts">
+	import { Modal, Button, PeriodicTable, Autocomplete } from 'svelte-spectre';
+	import elements from 'svelte-spectre/package/components/PeriodicTable/chemical_content.json';
+
+	import { DataView } from '@/views/modals';
+	import { createEventDispatcher } from 'svelte';
+
+	export let dataSource;
```

Comment:
@pyoner should this be initialized like above as `null`?

Reply:
no, because it's required property